libpod/config/config.go

```diff
@@ -0,0 +1,607 @@
+
+	// _conmonVersionFormatErr is used when the expected versio-format of conmon
+	// has changed.
+	_conmonVersionFormatErr = "conmon version changed format"
+
+	// _cgroupV2FallbackToCrun can be set at compile-time and allows for falling
+	// back to using crun when running on system with cgroupsv2. This is turned
+	// on by default and only used as a precaution - some distributions might
+	// not want it.
+	_cgroupV2FallbackToCrun = true
+)
+
+var (
+	// InstallPrefix is the prefix where podman will be installed.
+	// It can be overridden at build time.
+	installPrefix = "/usr"
+
+	// EtcDir is the sysconfdir where podman should look for system config files.
+	// It can be overridden at build time.
+	etcDir = "/etc"
+
+	// SeccompDefaultPath defines the default seccomp path.
+	SeccompDefaultPath = installPrefix + "/share/containers/seccomp.json"
+
+	// SeccompOverridePath if this exists it overrides the default seccomp path.
+	SeccompOverridePath = etcDir + "/crio/seccomp.json"
+
+	// _rootConfigPath is the path to the libpod configuration file
+	// This file is loaded to replace the builtin default config before
+	// runtime options (e.g. WithStorageConfig) are applied.
+	// If it is not present, the builtin default config is used instead
+	// This path can be overridden when the runtime is created by using
+	// NewRuntimeFromConfig() instead of NewRuntime().
+	_rootConfigPath = installPrefix + "/share/containers/libpod.conf"
+
+	// _rootOverrideConfigPath is the path to an override for the default libpod
+	// configuration file. If OverrideConfigPath exists, it will be used in
+	// place of the configuration file pointed to by ConfigPath.
+	_rootOverrideConfigPath = etcDir + "/containers/libpod.conf"
+)
+
+// Config contains configuration options used to set up a libpod runtime
+type Config struct {
+	// NOTE: when changing this struct, make sure to update (*Config).Merge().
+
+	// StateType is the type of the backing state store. Avoid using multiple
+	// values for this with the same containers/storage configuration on the
+	// same system. Different state types do not interact, and each will see a
+	// separate set of containers, which may cause conflicts in
+	// containers/storage. As such this is not exposed via the config file.
+	//
+	// TODO: this is runtime data and doesn't belong here.
+	StateType define.RuntimeStateStore `toml:"-"`
+	// Non-mergeable.
+
+	// StorageConfig is the configuration used by containers/storage Not
+	// included in the on-disk config, use the dedicated containers/storage
+	// configuration file instead.
+	StorageConfig storage.StoreOptions `toml:"-"`
+
+	// StorageConfigRunRootSet indicates if the RunRoot has been explicitly set
+	// by the config or by the user. It's required to guarantee backwards
+	// compatibility with older versions of libpod for which we must query the
+	// database configuration. Not included in the on-disk config.
+	StorageConfigRunRootSet bool `toml:"-"`
```

Comment:
Can we break this out to a sub-struct to group it together?

Reply:
Sounds good to me.